src/sandbox/interpolation/scripIO.cc

```diff
@@ -0,0 +1,264 @@
+        ATLAS_ASSERT(var_ia.getDimCount() == 1 && var_ia.getDim(0).getSize() == nnz);
+        ia.resize(nnz);  // NOTE: not compressed
+        var_ia.getVar(ia.data());
+
+        auto var_ja = f.getVar("src_address");
+        ATLAS_ASSERT(var_ja.getDimCount() == 1 && var_ja.getDim(0).getSize() == nnz);
+        ja.resize(nnz);
+        var_ja.getVar(ja.data());
+
+        auto var_a = f.getVar("remap_matrix");
+        ATLAS_ASSERT(var_a.getDimCount() == 2 && var_a.getDim(0).getSize() == nnz && var_a.getDim(1).getSize() == 1);
+        a.resize(nnz);
+        var_a.getVar(a.data());
+    }
+    catch (netCDF::exceptions::NcException& e) {
+        throw eckit::SeriousBug("SCRIP reading failed", Here());
+    }
+
+    // matrix conversion to 0-based indexing, sorting and compression
+    {
+        std::vector<size_t> sorted(nnz);
+        for (size_t n = 0; n < nnz; ++n) {
+            sorted[n] = n;
+        }
+
+        std::sort(sorted.begin(), sorted.end(), [&](auto i, auto j) {
+            return ia[i] != ia[j] ? ia[i] < ia[j] : ja[i] != ja[j] ? ja[i] < ja[j] : i < j;
+        });
+
+        decltype(ia) ia_unsorted(Nr + 1, 0);
+        decltype(ja) ja_unsorted(nnz);
+        ia.swap(ia_unsorted);
+        ja.swap(ja_unsorted);
+
+        const auto base = static_cast<Index>(1);
+        for (size_t n = 0; n < nnz; ++n) {
+            auto r = ia_unsorted[sorted[n]] - base;
+            auto c = ja_unsorted[sorted[n]] - base;
+            ATLAS_ASSERT(0 <= r && r < Nr);
+            ATLAS_ASSERT(0 <= c && c < Nc);
+
+            ia[r + 1]++;
+            ja[n] = c;
+        }
+
+        ia_unsorted.clear();
+        ja_unsorted.clear();
+
+        for (size_t r = 0; r < Nr; ++r) {
+            ia[r + 1] += ia[r];
+        }
+
+        ATLAS_ASSERT(ia.front() == 0 && ia.back() == nnz);
+
+        decltype(a) a_unsorted(nnz);
+        a.swap(a_unsorted);
+
+        for (size_t n = 0; n < nnz; ++n) {
+            a[n] = a_unsorted[sorted[n]];
+            ATLAS_ASSERT(0. <= a[n] && a[n] <= 1.);
+        }
+    }
+
+    eckit::linalg::SparseMatrix eckit_matrix(new InPlaceAllocator(Nr, Nc, nnz, ia.data(), ja.data(), a.data()));
+    return atlas::linalg::make_sparse_matrix_storage(eckit_matrix);
+#endif // ATLAS_HAVE_NETCDF
+}
+
+
+void ScripIO::write(const SparseMatrixStorage& matrix, const std::string& matrix_name) {
+#if ATLAS_HAVE_NETCDF == 0
+    ATLAS_THROW_EXCEPTION("Cannot write SCRIP files: Atlas not compiled with NetCDF support");
+#else
+    using scrip_index = int;
+    using scrip_value = double;
+    using scrip_size  = size_t;
+    scrip_size Nr  = matrix.rows();
+    scrip_size Nc  = matrix.cols();
+    scrip_size nnz = matrix.nnz();
+
+    std::vector<scrip_index> ia(nnz);
+    std::vector<scrip_index> ja(nnz);
+    std::vector<scrip_value> a(nnz);
+
+    matrix_to_rows_columns_values(matrix, ia, ja, a);
+    for (size_t i = 0; i < matrix.nnz(); ++i) {
+        ++ja[i];
+        ++ia[i];
+    }
```

Comment:
For intent and documentation:
```suggestion
    // SCRIP indices are 1-based
    constexpr scrip_index scrip_index_base = 1;
    for (size_t i = 0; i < matrix.nnz(); ++i) {
        ja[i] += scrip_index_base;
        ia[i] += scrip_index_base;
    }
```

Reply:
Done. Thanks!